openedx-unsupported/ecommerce
# REV-2687: update get_receipt_page_url to use new waffle flag

[REV-2687](https://2u-internal.atlassian.net/browse/REV-2687)

We want get_receipt_page_url to use a new waffle flag to determine whether to return the url for the original/live receipt page, or the new one being rebuilt in frontend-app-ecommerce in REV-2683. I've updated the function to take this parameter, and updated the places that call it to grab the value of the waffle flag and pass it along. There is one spot I didn't update: the send_mail function in EnrollmentCodeFulfillmentModule: it's an email that goes out when someone has purchased enrollment codes. In this case (or any where the waffle flag value is not passed), we default to False aka show the original ecommerce receipt page. 

## Testing instructions for this branch
- To start, place an order and see that afterward we're redirected to the old receipt page
- (then unenroll to be able to purchase that demo course again) 
- In your [local ecommerce admin page](http://localhost:18130/admin/waffle/flag/5/change/), add waffle flag set to enabled for everyone: use_new_receipt_page
- place an order again, and see that you're redirected to a different page (for now I'm using order history, but when the new receipt page is available we'll use that)

## Other information
- come back and confirm whether it's safe to assume this frontend-app-ecommerce url for everyone, vs make it something easier to configure


## Required Testing
(^ We can remove that manual check once REV-2624 is done and the corresponding e2e test runs again)
